text/0001-agile-coretime.md

```diff
@@ -0,0 +1,627 @@
+
+In every sale, a `BULK_LIMIT` of individual *Regions* are offered for sale.
+
+Each Region offered for sale has a different Core Index, ensuring that they each represent an independently allocatable resource on the Polkadot UC.
+
+The Regions offered for sale have the same span: they last exactly `BULK_PERIOD` blocks, and begin immediately following the span of the previous Sale's Regions. The Regions offered for sale also have the complete, non-interlaced, Core Mask.
+
+The Sale Period ends immediately as soon as span of the Coretime Regions that are being sold begins. At this point, the next Sale Price is set according to the previous Sale Price together with the number of Regions sold compared to the desired and maximum amount of Regions to be sold. See Price Setting for additional detail on this point.
+
+Following the end of the previous Sale Period, there is an *Interlude Period* lasting `INTERLUDE_PERIOD` of blocks. After this period is elapsed, regular purchasing begins with the *Purchasing Period*.
+
+This is designed to give at least two weeks worth of time for the purchased regions to be partitioned, interlaced, traded and allocated.
+
+#### The Interlude
+
+The Interlude period is a period prior to Regular Purchasing where renewals are allowed to happen. This has the effect of ensuring existing long-term tasks/parachains have a chance to secure their Bulk Coretime for a well-known price prior to general sales.
+
+#### Regular Purchasing
+
+Any account may purchase Regions of Bulk Coretime if they have the appropriate funds in place during the Purchasing Period, which is from `INTERLUDE_PERIOD` blocks after the end of the previous sale until the beginning of the Region of the Bulk Coretime which is for sale as long as there are Regions of Bulk Coretime left for sale (i.e. no more than `BULK_LIMIT` have already been sold in the Bulk Coretime Sale). The Purchasing Period is thus roughly `BULK_PERIOD - INTERLUDE_PERIOD` blocks in length.
+
+The Sale Price varies during an initial portion of the Purchasing Period called the *Leadin Period* and then stays stable for the remainder. This initial portion is `LEADIN_PERIOD` blocks in duration. During the Leadin Period the price decreases towards the Sale Price, which it lands at by the end of the Leadin Period. The actual curve by which the price starts and descends to the Sale Price is outside the scope of this RFC, though a basic suggestion is provided in the Price Setting Notes, below.
+
+#### Renewals
+
+At any time when there are remaining Regions of Bulk Coretime to be sold, *including during the Interlude Period*, then certain Bulk Coretime assignmnents may be *Renewed*. This is similar to a purchase in that funds must be paid and it consumes one of the Regions of Bulk Coretime which would otherwise be placed for purchase. However there are two key differences.
+
+Firstly, the price paid is exactly `RENEWAL_PRICE_CAP` more than what the purchase/renewal price was in the previous sale.
```

Comment:
Yeah, agreed - this mechanism would push renewals to go back onto the open market when the market price is falling. That could lead to worse guarantees of continuity, if demand happens to pick up again this period.

Reply:
I do think this is quite a bit worse than what was originally proposed by @gavofyork - that stated renewals should cost the lower of the current sale or current core cost multiplied by `RENEWAL_PRICE_CAP`.

Reply:
So as a workload which is renewable (practically speaking for now, a parachain, of group thereof), you would have three options:

- Pay 2% (or whatever) more than last month and guarantee your price and your slot.
- Pay the nominal (post-lead-in) price, which may be lower but requires competing within the market and waiting the lead-in period.
- Pay a higher-than-nominal price at some point during the lead-in period.

I would argue that placing a fixed-rate price increase could be defensible given that the buyer gets absolute certainty over availability. Nobody else in the market gets this. If the price ends up being higher than the apparent spot price, and the buyer wishes to optimize their costs, they are always free to join the regular market like everyone else.

The renewal mechanism is market-warping and has the potential to be anti-competitive. It should be subject to stringent safeguards.

Reply:
My problem with this is that to this date not a single parachain has done an automated trustless renewal, i.e. built a mechanism by which the parachain permissionlessly renewed a slot. The auctions are just too hard to build a strategy for and you're too much at a disadvantage as a parachain (vs. an offchain bidder). If we create a mechanism again where it's very hard for an existing workloads to renew at the optimal price then we will end up in the same situation again. 

If you are running a large application with many many users depending on it being available, **you will need certainty of uptime at all cost**.  So what would I do if I tried to design a strategy for Centrifuge? By default I will always have to renew because I can't risk not getting a core during LEADIN or post LEADIN.

Say prices are actually going down (which they should in the long term) I would then have to buy a second core on the market the minute it's cheap enough that over a few renewal periods I will break even paying for a core twice for one era. I would not want to risk going offline. This will lead again to very complex acquisition strategies which are not feasible to automate. We have to build a system that makes workload renewal very simple to automate and paid for by the chains themselves. **Everything else puts a huge amount of complexity onto teams that want to build in Polkadot.**

> The renewal mechanism is market-warping and has the potential to be anti-competitive. It should be subject to stringent safeguards.

Why not at least give renewals the same price as anyone could get by buying in the LEADIN period? I think this is a reasonable compromise.

That would satisfy two needs for existing workloads:
* Guarantee renewal provided the workload can pay *a fee*
* Guarantee that renewals are not more than the *highest possible* price in the current purchase period.

I'm sure some users will want to bet on a lower price than set in LEADIN and wait until after to bid but that would then come with the real risk of going offline.

Reply:
> Why not at least give renewals the same price as anyone could get by buying in the LEADIN period? I think this is a reasonable compromise.

Agree with this - I have already asked questions internally regarding how we could automate renewal for HydraDX. The benefit of the existing crowdloan model is we can almost forget about it for the best part of 2 years - suddenly requiring teams to be looking every 2-4 weeks for the optimal price is too much overhead.

Reply:
Probably a fair compromise.

Reply:
Changed to mirror this.